tests/requests/tests.py

```diff
@@ -778,6 +778,12 @@ def test_get_port_with_x_forwarded_port(self):
         }
         self.assertEqual(request.get_port(), '8080')
 
+    @override_settings(ALLOWED_HOSTS=['example.com'])
```

Comment:
I guess you'll have to test the `split_domain_port()` function since it'll be impossible to write a regression test using `HttpRequest.get_host()` since the fix is moving logic around in that method.